int16 limit1
int16 limit2
int16 limit3
int16 limit4
int16 limit5
int16 limit6
